core/src/main/java/com/sequenceiq/cloudbreak/conclusion/ConclusionChecker.java

```diff
@@ -25,25 +27,28 @@ public ConclusionChecker(List<ConclusionStep> conclusionSteps) {
 
     public ConclusionResult doCheck(Long resourceId) {
         LOGGER.info("Conclusion checker started, steps: {}, resourceId: {}", conclusionSteps, resourceId);
-        List<String> conclusions = new ArrayList<>();
+        List<Conclusion> conclusions = new ArrayList<>();
         try {
             while (hasMoreSteps() && doNext()) {
                 ConclusionStep conclusionStep = conclusionSteps.get(actualStep++);
                 LOGGER.debug("Conclusion step: {}", conclusionStep);
 
-                ConclusionStepResult stepResult = conclusionStep.check(resourceId);
-                LOGGER.debug("Conclusion step {}, conclusion: {}", stepResult.isStepFailed() ? "failed" : "succeeded", stepResult.getConclusion());
-                if (stepResult.getConclusion() != null) {
-                    conclusions.add(stepResult.getConclusion());
+                Conclusion conclusion = measure(() -> conclusionStep.check(resourceId),
+                        LOGGER, "Conclusion step finished in {} ms for resourceId {}, step {}", resourceId, conclusionStep.getClass().getSimpleName());
```

Comment:
Do we have a a CRN here? If it's included in the MdcContext that's fine, but the resourceId carries little information.

Reply:
MdcContext should contain related data.